qiskit_dynamics/signals/signals.py

```diff
@@ -494,12 +470,10 @@ def __len__(self):
         """Number of components."""
         return len(self.components)
 
-    def __getitem__(
-        self, idx: Union[int, List, np.array, slice]
-    ) -> Union[Signal, "SignalCollection"]:
+    def __getitem__(self, idx: Union[ArrayLike, slice]) -> Union[Signal, "SignalCollection"]:
         """Get item with NumPy-style subscripting, as if this class were a 1d array."""
 
-        if type(idx) == np.ndarray and idx.ndim > 0:
+        if type(idx) != int and type(idx) != slice and type(idx) != list and idx.ndim > 0:
```

Comment:
Hmm - the only issue with this current version is that when `idx.ndim > 0` is checked there is no guarantee that `idx` has an `ndim` attribute. I wonder if you can do:
```suggestion
        if type(idx) != slice and unp.asarray(idx).ndim > 0:
```
this will cover the `int` case, and we'll be calling `list` on a `list` for the `list` case, so that's fine too.